### ISSUE TITLE
Overriding default object loading doesn't not work if methods are private

### ISSUE THREAD

**Issue by tommmas** [stffn/declarative_authorization#62]
Hi,

When overriding the defaults object loading (in combination with filter_resource_access) I want to make the methods (such as 'load_user' etc.) private.

For this to work, the controller.respond_to? calls in lib/declarative_authorization/in_controller.rb need to have the second parameter set to true so that private methods are included in the search.

Has this just been overlooked or is there a reason to not allow these methods to be private?

**Comment by stffn** [comment]
No specific intentions there.  Any other side effects with private methods?  Have you tried it and perhaps even a test ready?

**Comment by tommmas** [comment]
I can't see any side effects of calling a private method in a controller filter. In fact, I think it is best practice for these methods to be private: if they are not, then they could be called directly by a request (if routing rules allow).

I've modified my local version of the gem and all is working as expected. I've had a look through the tests but can't find where this particular feature is tested. If you can point me to the right part I might be able to write a test.

Thanks for your great work on this gem. I've been using it on one of my production sites for some time with no issues.

**Comment by stffn** [comment]
Actually, if you are just trying to prevent the accidental calling of the load method by a request, you may also make them protected.  Still, I we should also allow private methods in addition.

Have you already forked the github repository?  You could push your changes there any make it very easy to have them incorporated into the main repository.

As for the tests, they belong to the controller_filter_resource_access_test.rb where there are already similar tests.

**Comment by tommmas** [comment]
Agreed, protected would work just as well in most cases. However, it seems fairly common to use private methods for this purpose and is even shown in the official docs (http://api.rubyonrails.org/classes/ActionController/Filters/ClassMethods.html). So allowing private methods will probably create least confusion.

**Comment by bluebird-communication** [comment]
As it is now 2 years past. Is an update of declarative_authorization planed to include this "private" feature stuff ?
